Sources/CorePayments/Networking/AnalyticsService.swift

```diff
@@ -49,7 +49,7 @@ public struct AnalyticsService {
         do {
             let clientID = coreConfig.clientID
             
-            let eventData = AnalyticsEventData(
+            let eventData = await AnalyticsEventData(
```

Comment:
q: This is an asynchronous constructor?

Reply:
Because AnalyticsEventData was marked with @MainActor, to instantiate it in async function that isn't isolated to the @MainActor, I needed "await".

Reply:
I used this async function and it passed. What do you think?
 ```
 func testVault_withValidResponse_returnsSuccess() async throws {
        let setupTokenID = "testToken1"
        let vaultStatus = "APPROVED"
        let vaultRequest = CardVaultRequest(card: card, setupTokenID: setupTokenID)
        let updateSetupTokenResponse = UpdateSetupTokenResponse(
            updateVaultSetupToken: TokenDetails(id: setupTokenID, status: vaultStatus, links: [TokenDetails.Link(rel: "df", href: "h")])
        )
        mockVaultAPI.stubSetupTokenResponse = updateSetupTokenResponse
        
        let expectation = expectation(description: "vault completed")
        let cardVaultDelegate = MockCardVaultDelegate(success: {_, result in
            XCTAssertEqual(result.setupTokenID, setupTokenID)
            XCTAssertEqual(result.status, vaultStatus)
            expectation.fulfill()
        }, error: {_, _ in
            XCTFail("Invoked error() callback. Should invoke success().")
        })
        sut.vaultDelegate = cardVaultDelegate
        sut.vault(vaultRequest)
        
        await self.fulfillment(of: [expectation], timeout: 10)
    }
```